src/net_processing.cpp

```diff
@@ -2795,6 +2808,61 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         return;
     }
 
+    // Received from an inbound peer planning to reconcile transactions with us, or
+    // from an outgoing peer demonstrating readiness to do reconciliations.
+    // If received from outgoing, adds the peer to the reconciliation queue.
+    // Feature negotiation of tx reconciliation should happen between VERSION and
+    // VERACK, to avoid relay problems from switching after a connection is up.
+    if (msg_type == NetMsgType::SENDRECON) {
+        if (!pfrom.m_tx_relay) return;
+        LOCK(peer->m_recon_state_mutex);
+        if (peer->m_recon_state != nullptr) return; // Do not support reconciliation salt/version updates.
+        LOCK(cs_main);
+        if (!State(pfrom.GetId())->m_wtxid_relay) return; // SENDRECON is allowed only after WTXIDRELAY.
+
+        bool recon_requestor, recon_responder;
+        uint64_t remote_salt;
+        uint32_t recon_version;
+        vRecv >> recon_requestor >> recon_responder >> recon_version >> remote_salt;
+        if (recon_version != 1) return;
+
+        // Do not flood through inbound connections which support reconciliation to save bandwidth.
+        bool flood_to = false;
+        if (pfrom.IsInboundConn()) {
+            // We currently don't support reconciliations with inbound peers which
+            // don't want to be reconciliation senders (request our sketches),
+            // or want to be reconciliation responders (send us their sketches).
+            // Just ignore SENDRECON and use normal flooding for transaction relay with them.
+            if (!recon_requestor) return;
+            if (recon_responder) return;
+        } else {
+            // We currently don't support reconciliations with outbound peers which
+            // don't want to be reconciliation responders (send us their sketches),
+            // or want to be reconciliation senders (request our sketches).
+            // Just ignore SENDRECON and use normal flooding for transaction relay with them.
+            if (recon_requestor) return;
+            if (!recon_responder) return;
+            // TODO: Flood only through a limited number of outbound connections.
+            flood_to = true;
+        }
+
+        uint64_t local_salt = peer->m_local_recon_salt;
+        uint64_t salt1 = local_salt, salt2 = remote_salt;
+        if (salt1 > salt2) std::swap(salt1, salt2);
+        static const auto RECON_SALT_HASHER = TaggedHash(RECON_STATIC_SALT);
+        uint256 full_salt = (CHashWriter(RECON_SALT_HASHER) << salt1 << salt2).GetSHA256();
+
+        peer->m_recon_state = MakeUnique<ReconState>(recon_requestor, recon_responder, flood_to, full_salt.GetUint64(0), full_salt.GetUint64(1));
+
+        // Reconcile with all outbound peers supporting reconciliation (even if we flood to them),
+        // to not miss transactions they have for us but won't flood.
```

Comment:
"strategic" is not a bidirectional link property, it's decided on every peer locally.
We might flood stuff to them, but they won't flood to us (because they only flood outbound). 

In that case, we will never get their transactions, if we don't reconcile.